### PR TITLE
UFAL/share submission by email

### LINKED ISSUE #463: Share submission by e-mail
**PRs:**
BE: https://github.com/dataquest-dev/DSpace/pull/780
FE: https://github.com/dataquest-dev/dspace-angular/pull/720
**Original issue:** https://github.com/ufal/clarin-dspace/issues/1044
**Excel notes:** `The user could share his submission with anyone else - Share submission button will be added`
1. Add a new `Share submission` button in the `Your submissions` view: 
2. Generate a link for sharing and store it in the workspaceitem table, column `sharetoken`
3. Create a new page `Share link` where will be showed the link for sharing with some info
4. Create tests
5. After clicking on the share link show Change submission page with the button `take it`. After pressing the button the submission owner is changed. Create tests

Steps:
1. Button: 
v5 Image
v7 Image

3. Share Submission with link page: 2.1. Show info with link Image
2.2. Send an email with link Image
3. Take it: Image

### PR DESCRIPTION
**Pull request by milanmajchrak**
| Phases            | MP | MM  |  MB  | MR |   JM | Total  |
|-----------------|----:|----:|----:|-----:|-----:|-------:|
| ETA                  |  0  |  0  |    0 |     0 |      0 |        0 |
| Developing      |  0  |  0  |    0 |    0 |      0 |         0 |
| Review             |  0  |  0  |    0 |    0 |      0 |         0 |
| Total                |   -  |   -  |   -   |  -    |   -    |         0 |
| ETA est.             |      |      |       |       |         |         0 |
| ETA cust.           |   -  |   -  |   -  |   -   |   -     |        0 |

### CODE REVIEW

File: src/app/change-submitter-page/change-submitter-page.component.ts [hunk 1]
```diff
@@ -0,0 +1,153 @@
...
+   * Load the submitter from the WorkspaceItem and assign it to the submitter BehaviorSubject.
+   */
+  loadAndAssignSubmitter(workspaceItem: WorkspaceItem) {
+    if (isNullOrUndef(workspaceItem)) {
+      console.error('Cannot load submitter because WorkspaceItem is null or undefined');
+      return;
+    }
+
+    if (workspaceItem.submitter instanceof Observable<EPerson>) {
+      console.log('Loading submitter from observable', workspaceItem);
```
Comment (vidiecan) [inline src/app/change-submitter-page/change-submitter-page.component.ts]: ?

File: src/app/change-submitter-page/change-submitter-page.component.ts [hunk 2]
```diff
@@ -0,0 +1,153 @@
...
+  /**
+   * Change the submitter of the WorkspaceItem using the shareToken. This will send a POST request to the backend when
+   * the submitter of the Item is changed.
+   */
+  changeSubmitter() {
+    const requestId = this.requestService.generateRequestId();
+
+    const url = this.halService.getRootHref() + '/submission/setOwner?shareToken=' + this.shareToken;
+    const postRequest = new PostRequest(requestId, url);
+    // Send GET request
```
Comment (vidiecan) [inline src/app/change-submitter-page/change-submitter-page.component.ts]: ?

File: src/app/shared/mydspace-actions/workspaceitem/workspaceitem-actions.component.ts [hunk 3]
```diff
@@ -46,8 +54,7 @@ export class WorkspaceitemActionsComponent extends MyDSpaceActionsComponent<Work
    * and therefore can delete it as well
    * (since the user can discard the item also from the edit page)
    * @type {Observable<boolean>}
-   */
-  canEditItem$: Observable<boolean>;
+   */  canEditItem$: Observable<boolean>;
```
Comment (vidiecan) [inline src/app/shared/mydspace-actions/workspaceitem/workspaceitem-actions.component.ts]: ?

**Review by vidiecan — CHANGES_REQUESTED** [review]

File: src/app/change-submitter-page/change-submitter-page.component.ts (hunk 2 above)
Comment (milanmajchrak) [reply src/app/change-submitter-page/change-submitter-page.component.ts:135]: Removed, thank you

File: src/app/change-submitter-page/change-submitter-page.component.ts (hunk 1 above)
Comment (milanmajchrak) [reply src/app/change-submitter-page/change-submitter-page.component.ts]: Removed, thank you

File: src/app/shared/mydspace-actions/workspaceitem/workspaceitem-actions.component.ts (hunk 3 above)
Comment (milanmajchrak) [reply src/app/shared/mydspace-actions/workspaceitem/workspaceitem-actions.component.ts]: Fixed

**Review by vidiecan — APPROVED** [review]

**Pull request merged**